### PR TITLE
Apply GitHub authentication plugin

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add a plugin that authenticates [vscode GitHub PR plugin](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-pull-request-github) launched in Che-Theia
ezgif-2-e8952916c88e

The plugin calls Che oAuth API service to get the GitHub token. Then it injects it to the user preferences. When the browser page is refreshed the vscode GitHub PR plugin fetches the token and cleans the token from the preferences file.

This works only in single user Che. When https://github.com/eclipse/che/issues/15672 is applied the token could be fetched in the multi-user.

**What can be improved:**
The login actions that provided by the native vscode GitHub plugin are still redirect to the vscode authentication service. To override them we need:
1.  Apply `Urihandler` plugin API method: https://github.com/eclipse-theia/theia/issues/5119. It is used by the vscode GitHub PR plugin to handle the callback request from authentication server.
2. Authentication server url is hard-coded in the vscode GitHub PR plugin: https://github.com/microsoft/vscode-pull-request-github/blob/ddaa2bcd06ebc931e075cc9cab5a83f683e93e83/src/authentication/githubServer.ts#L16-L17
https://github.com/microsoft/vscode-pull-request-github/blob/ddaa2bcd06ebc931e075cc9cab5a83f683e93e83/src/authentication/githubServer.ts#L177-L179
so we need to propose a PR to the [vscode GitHub PR plugin](https://github.com/Microsoft/vscode-pull-request-github) repository that allows to override the authentication urls.

When the steps above are done we can setup the vscode GitHub PR plugin to use Che oAuth service for authentication and get rid of this authentication plugin.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14217

#### Docs PR

https://github.com/eclipse/che-docs/pull/1055

### CODE REVIEW

File: plugins/github-auth-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Theia - Telemetry Plug-in
```
Comment (RomanNikitenko) [inline plugins/github-auth-plugin/README.md]: something wrong here :-)

Comment (vinokurig) [reply plugins/github-auth-plugin/README.md]: Thanks, fixed the description :)

File: plugins/github-auth-plugin/src/github-auth-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import * as che from '@eclipse-che/plugin';
+
+export function start(context: theia.PluginContext) {
+    if (theia.plugins.getPlugin('github.vscode-pull-request-github')) {
+        const informationMessageTestCommand = {
```
Comment (benoitf) [inline plugins/github-auth-plugin/src/github-auth-plugin.ts]: looks the variable name is not self explanatory

**Comment by benoitf** [comment]
hello, why do we need to refresh the browser ? (on UX it may not be so nice to do that)

File: plugins/github-auth-plugin/src/github-auth-plugin.ts (hunk 2 above)
Comment (vinokurig) [reply plugins/github-auth-plugin/src/github-auth-plugin.ts]: fixed

**Comment by vinokurig** [comment]
@benoitf I agree that it is not a good UX, but the vscode GitHub PR plugin loads the token from configuration on it's initialisation https://github.com/microsoft/vscode-pull-request-github/blob/6b019aa02b90cc5729f14ba8b98f653b34fa64a6/src/extension.ts#L37

File: plugins/github-auth-plugin/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+      "name": "@eclipse-che/github-auth-plugin",
+      "version": "0.0.1",
+      "publisher": "Eclipse Che",
+      "keywords": [
+        "theia-plugin"
+      ],
+      "description": "Manage Eclipse Che Telemetry Plugin",
```
Comment (RomanNikitenko) [inline plugins/github-auth-plugin/package.json]: please fix the description

File: plugins/github-auth-plugin/src/github-auth-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
```
Comment (RomanNikitenko) [inline plugins/github-auth-plugin/src/github-auth-plugin.ts]: 2020?

File: plugins/github-auth-plugin/package.json (hunk 3 above)
Comment (vinokurig) [reply plugins/github-auth-plugin/package.json]: done

File: plugins/github-auth-plugin/src/github-auth-plugin.ts (hunk 4 above)
Comment (vinokurig) [reply plugins/github-auth-plugin/src/github-auth-plugin.ts]: done

**Comment by vinokurig** [comment]
crw-ci-test

**Pull request merged**